x/ibc/07-tendermint/types/client_state.go

```diff
@@ -158,6 +177,8 @@ func (cs ClientState) VerifyClientConsensusState(
 		return err
 	}
 
+	// TODO: Inject proof specs of clientstate into proof or modify verify membership function to take proof specs
```

Comment:
As @ethanfrey mentioned, the proof spec needs to be fixed by the client not passed in by the user. I can either inject the clientstate proofspecs into the `MerkleProof` struct (replacing whatever user passed in). Or i can remove specs from `MerkleProof` and pass it into the `VerifyMembership function`.

The latter requires a change in ICS spec

Reply:
We should change ICS 23 to not put `ProofSpec` in the proofs

If this requires a spec change as well, we can do that